src/core/sock/sock-extra.cpp

```diff
@@ -406,6 +406,9 @@ extern "C" int xlio_init_ex(const struct xlio_init_attr *attr)
         setenv(SYS_VAR_PROGRESS_ENGINE_INTERVAL, "0", 1);
     }
 
+    // Read the updated parameters. A global object could trigger the reading earlier.
+    safe_mce_sys().get_env_params();
```

Comment:
Maybe a note to XLIO Gaps: We need to remove any possibly unordered initialization in global constructors and do initialization in a controlled manner..

Reply:
will add a note in the document